src/core/layertree/qgslayertreemodellegendnode.h

```diff
@@ -306,6 +338,15 @@ class CORE_EXPORT QgsSymbolLegendNode : public QgsLayerTreeModelLegendNode
   private:
     void updateLabel();
 
+    /**
+    * Evaluate the label with the proper context
+    * \param label text to evaluate
+    * \param context extra QgsExpressionContext to use for evaluation
+    * \since QGIS 3.8
+    */
+    QString evaluateLabelExpression( const QString &label, QgsExpressionContext &context ) const;
```

Comment:
```suggestion
    QString evaluateLabelExpression( const QString &label, const QgsExpressionContext &context ) const;
```

Reply:
I cannot do this since the symbol scope is appended to the context in the function.